lib/public/init.js

```diff
@@ -27,6 +27,16 @@ module.exports = function (browserSync, name, pjson) {
 
         args.config.version = pjson.version;
 
+        /**
+         * Preserve the httpModule property's functions.
+         * the http2 module exports an object of functions and the merge function seems
+         * to want to destroy that, but if the base object is a function it seems fine
+         * TODO: find a better or more generic way to handle this
+         */
+        if(args.config.httpModule && !_.isFunction(args.config.httpModule)) {
+            args.config.httpModule = Object.assign(function() {}, args.config.httpModule);
+        }
```

Comment:
@itslenny So if I want to use this in my gulpfile with a proxy, I could write
`browserSync.init({ proxy: url, httpModule: http2 });` ?